src/toc.tsx

```diff
@@ -99,7 +102,31 @@ export class TableOfContents extends Widget {
         title = PathExt.basename(context.localPath);
       }
     }
-    ReactDOM.render(<TOCTree title={title} toc={toc} />, this.node, () => {
+    let itemRenderer: (item: IHeading) => JSX.Element | null = (
+      item: IHeading
+    ) => {
+      return <span>{item.text}</span>;
+    };
+    if (this._current && this._current!.generator.itemRenderer) {
```

Comment:
I don't think the `!`s are necessary here, since you are already checking for existence. Could be wrong, though.